contrib/devtools/lint-extern-fundecl.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Check unnecessary extern keyword usage for function declarations.
+
+HEADER_ID_PREFIX="BITCOIN_"
+HEADER_ID_SUFFIX="_H"
+
+REGEXP_EXCLUDE_FILES_WITH_PREFIX="src/(crypto/ctaes/|leveldb/|secp256k1/|tinyformat.h|univalue/)"
+
+EXIT_CODE=0
+for SOURCE_FILE in $(git ls-files -- "*.h" "*.cpp" | grep -vE "^${REGEXP_EXCLUDE_FILES_WITH_PREFIX}")
+do
+    if [[ $(egrep -c "extern ([^_\"\(]+)\(" ${SOURCE_FILE}) != 0 ]]; then
```

Comment:
`shellcheck: SC2196: egrep is non-standard and deprecated. Use grep -E instead.`

`shellcheck: SC2086: Double quote to prevent globbing and word splitting.` (`"${SOURCE_FILE}" instead of ${SOURCE_FILE}`)